### PR TITLE
Documentation: Jsdoc-convert refactoring

### PR DESCRIPTION
**Pull request by Siemienik**
### Context

**THIS PR BLOCKS: #8607 and #8609**

I started resolving #8335, however, it requires pre-refactoring before do it well.

So, in this PR I refactored Jsdoc-convert tool, It does not change the logic of how the output is generated. So it still generates the same output.
    
I introduced inversion of control to manage dependencies between files, that makes dependency flow flatter.
Then, I describe the flow in the Readme as a plantuml diagram.

Here:    
* I extracted configuration - what in my opinion is needed to prevent changing the jsdoc-convert logic by a mistake further.
* I extracted the integration layer (jsdoc-to-markdown) - which makes the external dependency a little weaker.
* I extracted post and preprocessors, thanks to this it is much more self-described.
* I extracted smaller modules like paths, seo, and predictors to keep the file structure robust.

I didn't refactor part of jsdoc convert which I plan to change during resolving the #8335 issue.

_[skip changelog]_

### How has this been tested?
I generate API Refs files and compare them with those already existed in v9.0.

### Types of changes
- [x] New feature or improvement 

### Related issue(s):
1. #8335  (do not close this issue)
2.
3.

### Checklist:
- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)

### CODE REVIEW

**Comment by swistach** [comment]
Are you sure you want to merge to that base branch? :)

File: docs/.vuepress/tools/jsdoc-convert/configuration.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+module.exports = {
```
Comment (budnix) [inline docs/.vuepress/tools/jsdoc-convert/configuration.js:1]: Is there is a reason to keep the file in CommonJS format? All files within the module are refactored to `mjs` this is the last one.

File: docs/.vuepress/tools/jsdoc-convert/index.mjs [hunk 2]
```diff
@@ -0,0 +1,107 @@
+/* eslint-disable no-useless-escape */
```
Comment (budnix) [inline docs/.vuepress/tools/jsdoc-convert/index.mjs:1]:
```suggestion
```
I think that if the Eslint rule says that there is a useless regex escape character we can remove that. WDYT?

File: docs/.vuepress/tools/jsdoc-convert/integrations/jsdoc-to-markdown/index.mjs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import jsdoc2md from 'jsdoc-to-markdown';
+import dmd from 'dmd';
+import path from 'path';
+import {fileURLToPath} from "url";
```
Comment (budnix) [inline docs/.vuepress/tools/jsdoc-convert/integrations/jsdoc-to-markdown/index.mjs:4]: This should be caught by the ESlint. Can you update the npm scripts and include the `.mjs` files in the lining process?

```
"docs:lint": "eslint --ext .js,.vue,.mjs .vuepress next",
"docs:lint:fix": "eslint --ext .js,.vue,.mjs .vuepress next --fix",
```

**Review by budnix — CHANGES_REQUESTED** [review]
Nice refactor 👍  With that small functions we are closer to write unit tests to them.

**Review by aninde — DISMISSED** [review]
👍

File: docs/.vuepress/tools/jsdoc-convert/configuration.js (hunk 1 above)
Comment (Siemienik) [reply docs/.vuepress/tools/jsdoc-convert/configuration.js:1]: That way it is closer to json format, which I would like to use here. However, it doesn't work for me.

Comment (budnix) [reply docs/.vuepress/tools/jsdoc-convert/configuration.js:1]: You can export the config via defult export, like:
```js
const config = {
  pathToSource: '../../../../src',
  pathToDist: '../../../next/api',
  urlPrefix: '/next/api/',
  whitelist: [
    'dataMap/metaManager/metaSchema.js',
    'pluginHooks.js',
    'core.js',
    'translations/indexMapper.js',
    'editors/baseEditor/baseEditor.js',
    '3rdparty/walkontable/src/cell/coords.js',
    'plugins/copyPaste/focusableElement.js',
    'dataMap.js',
    'translations/maps/hidingMap.js',
    'translations/maps/indexesSequence.js',
    'translations/maps/trimmingMap.js',
    'utils/samplesGenerator.js',
    'translations/maps/physicalIndexToValueMap.js',
    'utils/ghostTable.js'
  ],
  seo: {
    'dataMap/metaManager/metaSchema.js': {
      title: 'Options',
      metaTitle: 'Options - API Reference - Handsontable Documentation',
      permalink: '/next/api/options'
    },
    'pluginHooks.js': {
      title: 'Hooks',
      metaTitle: 'Hooks - API Reference - Handsontable Documentation',
      permalink: '/next/api/hooks'
    },
    'core.js': {
      title: 'Core',
      metaTitle: 'Core - API Reference - Handsontable Documentation',
      permalink: '/next/api/core'
    },
    '3rdparty/walkontable/src/cell/coords.js': {
      title: 'CellCoords',
      metaTitle: 'CellCoords - API Reference - Handsontable Documentation',
      permalink: '/next/api/coords'
    }
  },
  linkAliases: {
    options: 'metaSchema',
    hooks: 'pluginHooks'
  }
};

export default config;
```
Then I run `npm run docs:api`. I haven't noticed any issues.

File: docs/.vuepress/tools/jsdoc-convert/integrations/jsdoc-to-markdown/index.mjs (hunk 3 above)
Comment (Siemienik) [reply docs/.vuepress/tools/jsdoc-convert/integrations/jsdoc-to-markdown/index.mjs]: Good catch, done bc8d959

**Review by budnix — CHANGES_REQUESTED** [review]
Can I ask to resolve the conflicts before my review? And what do you think about https://github.com/handsontable/handsontable/pull/8590#discussion_r682354954?

File: docs/.vuepress/tools/jsdoc-convert/configuration.js (hunk 1 above)
Comment (Siemienik) [reply docs/.vuepress/tools/jsdoc-convert/configuration.js]: done 436f5ca4ca416de4c942f851a9d2476c08e3089a

**Review by budnix — APPROVED** [review]
👍

**Review by aninde — APPROVED** [review]
Reapprove :)

**Pull request merged**